ibm/data_source_ibm_is_security_group_rule_test.go

```diff
@@ -0,0 +1,114 @@
+// Copyright IBM Corp. 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package ibm
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccIBMIsSecurityGroupRuleDataSourceBasic(t *testing.T) {
+	vpcname := fmt.Sprintf("tfsecgrprl-vpc-%d", acctest.RandIntRange(10, 100))
+	sgname := fmt.Sprintf("tfsecgrprl-name-%d", acctest.RandIntRange(10, 100))
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckIBMIsSecurityGroupRuleDataSourceConfigBasic(vpcname, sgname),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("data.ibm_is_security_group_rule.is_security_group_rule", "security_group"),
+					resource.TestCheckResourceAttrSet("data.ibm_is_security_group_rule.is_security_group_rule", "direction"),
+					resource.TestCheckResourceAttrSet("data.ibm_is_security_group_rule.is_security_group_rule", "href"),
+					resource.TestCheckResourceAttrSet("data.ibm_is_security_group_rule.is_security_group_rule", "ip_version"),
+					resource.TestCheckResourceAttrSet("data.ibm_is_security_group_rule.is_security_group_rule", "protocol"),
+					resource.TestCheckResourceAttrSet("data.ibm_is_security_group_rule.is_security_group_rule", "remote.#"),
+				),
+			},
+		},
+	})
+}
+
+// func testAccCheckIBMIsSecurityGroupRuleDataSourceConfigBasic(vpcname, sgname string) string {
+// 	return fmt.Sprintf(`
+// 	data "ibm_is_security_group_rule" "is_security_group_rule" {
+// 					// security_group = ibm_is_security_group.testacc_security_group.id
+// 					// security_group_rule = ibm_is_security_group_rule.testacc_security_group_rule_all.id
+// 					security_group_rule = "r134-5b206956-0990-4337-bab4-02005315db0a"
+// 					security_group = "r134-4d5db1a7-ca45-4c29-80d1-2a90cf94d11a"
+// 				}
+// 	`)
+// }
+
+func testAccCheckIBMIsSecurityGroupRuleDataSourceConfigBasic(vpcname, sgname string) string {
+	return fmt.Sprintf(`
+
+	resource "ibm_is_vpc" "testacc_vpc" {
+        name = "%s"
+      }
+      
+      resource "ibm_is_security_group" "testacc_security_group" {
+        name = "%s"
+        vpc  = ibm_is_vpc.testacc_vpc.id
+      }
+      
+      resource "ibm_is_security_group_rule" "testacc_security_group_rule_all" {
```

Comment:
No need to create 4 rules as we are having only one data source, it is enough to create just one rule resource.

Reply:
done.